libs/ds18b20/ds18b20.cpp

```diff
@@ -1,19 +1,119 @@
 #include "ds18b20.h"
 
+/**
+ * If DS18B20 resolution is below maximum, the omitted lower bits are
+ * undefined/floating. Therefore, the typical errors (nc: -127 C, conversion
+ * error: +85 C) may not occur exactly. The maximal offset DS18B20_ERROR_TOL
+ * can be chosen as small as possible, depending on the set resolution.
+ *  9 bit: -127.4375 to -127.0000   and   +85.0000 to +85.4375
+ * 10 bit: -127.1875 to -127.0000   and   +85.0000 to +85.1875
+ * 11 bit: -127.0625 to -127.0000   and   +85.0000 to +85.0625
+ * 12 bit:               127.0000   and   +85.0000
+ */
+#define DS18B20_DISCONNECTED (-127.0)
+#define DS18B20_CONVERSION_ERROR (85.0)
+
+#define DS18B20_RESOLUTION 10       // set desired resolution in bit (9 to 12)
+
+/**
+ * Define error tolerance from resolution, chosen slightly ceiled.
+ * Since neither DallasTemperature functions 'isConnected' nor
+ * 'isConversionComplete' work in parasite mode, conversion time has to
+ * be considere explicitly.
+ * 9-12 bit resolution: 94, 188, 375, 750 ms conversion time
+ */
+#if DS18B20_RESOLUTION == 9
+#define DS18B20_ERROR_TOLERANCE 0.45
+#define DS18B20_CONVERSION_DURATION 94
+#elif DS18B20_RESOLUTION == 10
+#define DS18B20_ERROR_TOLERANCE 0.20
+#define DS18B20_CONVERSION_DURATION 188
```

Comment:
You mean remove all but the 10bit values?

I already refrained from adding the resolution to `./configure`. Here, I think its convenient, because I only need to update one value instead of 3, if I ever wanted to change the resolution. Though I have to admit that I do not know what happens if someone gets the crazy idea to add one of these definitions to `config.h`.

Reply:
> You mean remove all but the 10bit values?

Yes. Ideally if something is configurable through `./configure` and a configuration we know exactly what the controller runs.

> if I ever wanted to change the resolution

That's my point entirely. If we see the need to have an option, okay. But if not, it's just unnecessary.

Reply:
Amended in the latest version.